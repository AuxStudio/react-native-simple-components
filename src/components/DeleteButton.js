import React from 'react';
import { View, StyleSheet } from 'react-native';

import Icon from '../assets/icons/index';
import styleConstants from '../assets/styleConstants';

import Touchable from './Touchable';

const styles = StyleSheet.create({
    deleteButtonContainer: {},
    deleteButton: {
        backgroundColor: styleConstants.lightGrey,
        width: 20,
        height: 20,
        borderRadius: 10,
        justifyContent: 'center',
        alignItems: 'center',
    },
    deleteIcon: {
        fontSize: 12,
        color: styleConstants.primary,
    },
});

export default (DeleteButton = props => {
    /*
        PROPTYPES

        handlePress: PropTypes.func.isRequired,
    */

    return (
        <View style={styles.deleteButtonContainer}>
            <Touchable onPress={props.handlePress} style={styles.deleteButton}>
                <Icon name="close" style={styles.deleteIcon} />
            </Touchable>
        </View>
    );
});
