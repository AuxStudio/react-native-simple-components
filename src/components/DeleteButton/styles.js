import { StyleSheet } from 'react-native';

import styleConstants from '../../styleConstants';

const styles = StyleSheet.create({
  container: {
    width: 20,
    height: 20,
    borderRadius: 10,
    overflow: 'hidden',
    borderWidth: 0,
    backgroundColor: styleConstants.dividerColor,
  },
  icon: {
    fontSize: styleConstants.smallFont,
    color: styleConstants.primaryText,
  },
});

export default styles;
