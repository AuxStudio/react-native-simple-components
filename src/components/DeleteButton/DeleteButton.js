import React from 'react';

import styles from './styles';
import styleConstants from '../assets/styleConstants';

import ButtonIcon from './ButtonIcon';

export default (DeleteButton = (props) => {
  /*
        PROPTYPES

        handlePress: PropTypes.func.isRequired,

        // iconStyle: PropTypes.node,
        // style: PropTypes.node,
    */

  return (
    <ButtonIcon
      handlePress={props.handlePress}
      iconName="close"
      iconStyle={[styles.icon, props.iconStyle]}
      style={[styles.container, props.style]}
    />
  );
});
