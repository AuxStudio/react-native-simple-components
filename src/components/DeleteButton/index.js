import React from 'react';
import PropTypes from 'prop-types';
import { ViewPropTypes, Text } from 'react-native';

import styles from './styles';
import styleConstants from '../../styleConstants';
import ButtonIcon from '../ButtonIcon';

const propTypes = {
  handlePress: PropTypes.func,
  customIcon: PropTypes.node,
  showShadow: PropTypes.bool,
  androidRipple: PropTypes.bool,
  androidRippleColor: PropTypes.string,
  androidRippleBorderless: PropTypes.bool,
  iconStyle: Text.propTypes.style,
  style: ViewPropTypes.style,
};

const defaultProps = {};

const DeleteButton = ({
  handlePress,
  customIcon,
  showShadow,
  androidRipple,
  androidRippleColor,
  androidRippleBorderless,
  iconStyle,
  style,
}) => {
  const shadowStyles = showShadow && styleConstants.shadows.small;

  return (
    <ButtonIcon
      handlePress={handlePress}
      iconName="close"
      customIcon={customIcon}
      androidRipple={androidRipple}
      androidRippleColor={androidRippleColor}
      androidRippleBorderless={androidRippleBorderless}
      iconStyle={[styles.icon, iconStyle]}
      style={[styles.container, shadowStyles, style]}
    />
  );
};

DeleteButton.propTypes = propTypes;
DeleteButton.defaultProps = defaultProps;

export default DeleteButton;
