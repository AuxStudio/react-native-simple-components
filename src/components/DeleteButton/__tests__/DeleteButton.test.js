import React from 'react';
import renderer from 'react-test-renderer';
import MaterialIcon from 'react-native-vector-icons/MaterialIcons';

import DeleteButton from '../';

describe('DeleteButton', () => {
  it('renders with all props', () => {
    expect(
      renderer.create(
        <DeleteButton
          handlePress={jest.fn()}
          showShadow
          androidRipple
          androidRippleColor="red"
          androidRippleBorderless
          iconStyle={{ color: 'red' }}
          style={{ backgroundColor: 'blue' }}
        />,
      ),
    ).toMatchSnapshot();
  });

  it('renders with minimum required props', () => {
    expect(renderer.create(<DeleteButton />)).toMatchSnapshot();
  });

  it('renders with custom icon', () => {
    expect(
      renderer.create(<DeleteButton customIcon={<MaterialIcon name="check" />} />),
    ).toMatchSnapshot();
  });
});
