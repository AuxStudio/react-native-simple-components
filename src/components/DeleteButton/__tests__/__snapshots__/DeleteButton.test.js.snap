// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteButton renders with all props 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "blue",
      "borderColor": null,
      "borderRadius": 10,
      "borderWidth": 0,
      "elevation": 2,
      "height": 20,
      "justifyContent": "center",
      "opacity": 1,
      "position": "relative",
      "shadowColor": "#212121",
      "shadowOffset": Object {
        "height": 1,
        "width": 0,
      },
      "shadowOpacity": 0.17,
      "shadowRadius": 2,
      "width": 20,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": undefined,
          "fontSize": 12,
        },
        Array [
          Object {
            "color": "#212121",
            "fontSize": 24,
          },
          Array [
            Object {
              "color": "#212121",
              "fontSize": 15,
            },
            Object {
              "color": "red",
            },
          ],
        ],
        Object {
          "fontFamily": "Material Icons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
      ]
    }
  >
    
  </Text>
</View>
`;

exports[`DeleteButton renders with custom icon 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#E0E0E0",
      "borderRadius": 10,
      "borderWidth": 0,
      "height": 20,
      "justifyContent": "center",
      "opacity": 1,
      "position": "relative",
      "width": 20,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": undefined,
          "fontSize": 12,
        },
        undefined,
        Object {
          "fontFamily": "Material Icons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
      ]
    }
  >
    
  </Text>
</View>
`;

exports[`DeleteButton renders with minimum required props 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#E0E0E0",
      "borderRadius": 10,
      "borderWidth": 0,
      "height": 20,
      "justifyContent": "center",
      "opacity": 1,
      "position": "relative",
      "width": 20,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": undefined,
          "fontSize": 12,
        },
        Array [
          Object {
            "color": "#212121",
            "fontSize": 24,
          },
          Array [
            Object {
              "color": "#212121",
              "fontSize": 15,
            },
            undefined,
          ],
        ],
        Object {
          "fontFamily": "Material Icons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
      ]
    }
  >
    
  </Text>
</View>
`;
